test/ConstExtraction/ExtractKeyPaths.swift

```diff
@@ -0,0 +1,75 @@
+// RUN: %empty-directory(%t)
+// RUN: echo "[MyProto]" > %t/protocols.json
+
+// RUN: %target-swift-frontend -typecheck -emit-const-values-path %t/ExtractTypes.swiftconstvalues -const-gather-protocols-file %t/protocols.json -primary-file %s
+// RUN: cat %t/ExtractTypes.swiftconstvalues 2>&1 | %FileCheck %s
+
+protocol MyProto {}
+
+public struct MyType {
+    var nested: NestedOne
+
+    struct NestedOne {
+        var foo: NestedTwo
+    }
+
+    struct NestedTwo {
+        var bar: NestedThree
+    }
+
+    struct NestedThree {
+        var baz: String
+    }
+}
+
+public struct KeyPaths: MyProto {
+    static let nestedVariable = \MyType.nested.foo.bar.baz
+}
+
+// CHECK: [
+// CHECK-NEXT:   {
+// CHECK-NEXT:     "typeName": "ExtractKeyPaths.KeyPaths",
+// CHECK-NEXT:     "mangledTypeName": "15ExtractKeyPaths0bC0V",
+// CHECK-NEXT:     "kind": "struct",
+// CHECK-NEXT:     "file": "SOURCE_DIR/test/ConstExtraction/ExtractKeyPaths.swift",
```

Comment:
To have these tests work on windows, I believe instead of `/` you need to specify `{{/|\\}}`.

Reply:
Ok changed that and restarted the tests.